lib/Interpreter/Transaction.cpp

```diff
@@ -67,11 +67,12 @@ namespace cling {
     for (auto I = decls_begin(), E = decls_end(); I < E; ++I) {
       for (auto DI : I->m_DGR) {
         if (DI->getKind() == Decl::LinkageSpec) {
-          for ( Decl *DI : LinkageSpecDecl::castToDeclContext(
-                static_cast<LinkageSpecDecl*>(DI))->decls()) {
-            if (NamedDecl* ND = dyn_cast<NamedDecl>(DI)) {
-              if (name.equals(ND->getNameAsString()))
-                return ND;
+          if (LinkageSpecDecl* LSD = cast<LinkageSpecDecl>(DI)) {
```

Comment:
On 15/06/16 18:58, marsupial wrote:
>
> In lib/Interpreter/Transaction.cpp 
> <https://github.com/root-mirror/cling/pull/17#discussion_r67201660>:
>
> > @@ -67,11 +67,12 @@ namespace cling {
> >      for (auto I = decls_begin(), E = decls_end(); I < E; ++I) {
> >        for (auto DI : I->m_DGR) {
> >          if (DI->getKind() == Decl::LinkageSpec) {
> > -          for ( Decl *DI : LinkageSpecDecl::castToDeclContext(
> > -                static_cast<LinkageSpecDecl*>(DI))->decls()) {
> > -            if (NamedDecl* ND = dyn_cast<NamedDecl>(DI)) {
> > -              if (name.equals(ND->getNameAsString()))
> > -                return ND;
> > +          if (LinkageSpecDecl* LSD = cast<LinkageSpecDecl>(DI)) {
>
> But the first check is to avoid diving of into the type-system on 
> every decl (which will likely be many). I think it's a bit more 
> efficient this way, but if you think otherwise I'll concede.
>
I prefer the dyn_cast to go in the place of the DI->getKind() and then 
the if stmt containing the cast will go away.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/root-mirror/cling/pull/17/files/e135743361fe83b6f9d67c361be5099f1f4fd0fe..a65a6d7209e3b8151134c7f3357199fbd9af5d34#r67201660>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/AGNuUxwDCIH4OvFpxaOTTO-2PLC4Ei8pks5qMC8zgaJpZM4I0etZ>.
>